ltm/inserted_context_agent.py

```diff
@@ -0,0 +1,577 @@
+
+from utils.llm import LLMContext, make_system_message, make_user_message, make_assistant_message, ensure_context_len, \
+    ask_llm
+from utils.text import truncate
+from utils.ui import colour_print
+
+
+def dump_context_s(context):
+    messages = []
+    for message in context:
+        messages.append(f"---\nRole: {message['role']}\nContent:\n{message['content']}")
+
+    return "\n".join(messages)
+
+@dataclass
+class InsertedContextAgent:
+    max_prompt_size: int
+    max_completion_tokens: Optional[int] = None
+    semantic_memory: DefaultTextMemory = field(default_factory=AutoTextMemory.create)
+    max_prompt_size: int = 16384
+    is_local: bool = True
+    defined_kws: list = field(default_factory=list)
+    scratchpad: dict[str, str] = field(default_factory=dict)
+    scratchpad_ts: dict[str, float] = field(default_factory=dict)  # Tracks changes in user_info (timestamps)
+    llm_index: int = 0
+    costs_usd: float = 0.0
+    model: str = "together_ai/meta-llama/Meta-Llama-3.1-70B-Instruct-Turbo"
+    temperature: float = 0.01
+    system_message = """You are a helpful AI assistant."""
+
+    def __post_init__(self):
+        self.semantic_memory = AutoTextMemory.create(config=TextMemoryConfig(chunk_capacity=50, chunk_overlap_fraction=0.0))
+        self.is_local = True
+        self.max_message_size = 1000
+        self.defined_kws = []
+        self.new_session()
+
+    def new_session(self) -> 'LTMAgentSession':
+        """
+        Creates a new LTMAgentSession object and sets it as the current session.
+        :return: The new session object.
+        """
+        session_id = str(uuid.uuid4())
+        self.session = LTMAgentSession(session_id=session_id, m_history=[], i_dict={})
+        if not self.semantic_memory.is_empty():
+            self.semantic_memory.add_separator()
+        return self.session
+
+    def reply(self, user_message: str, agent_response: Optional[str] = None, cost_callback: Callable=None) -> str:
+
+        colour_print("CYAN", f"DEALING WITH USER MESSAGE: {user_message}")
+
+        keywords = self.keywords_for_message(user_message, cost_cb=cost_callback)
+        context = self.create_context(user_message, max_prompt_size=self.max_prompt_size, previous_interactions=0, cost_cb=cost_callback)
+        if not os.path.exists(f"data/llm_calls/{self.session.session_id}"):
+            os.mkdir(f"data/llm_calls/{self.session.session_id}")
+
+        fname = f"data/llm_calls/{self.session.session_id}/call_{self.llm_index}.json"
+        with open(fname, "w") as f:
+            colour_print("BlUE", f"Saving to: {fname}")
+            f.write(dump_context_s(context))
+            response_text = ask_llm(context, model=self.model, max_overall_tokens=self.max_prompt_size,
+                                    cost_callback=cost_callback, temperature=self.temperature)
+
+            f.write(f"\n\nResponse:\n{response_text}")
+            self.llm_index += 1
```

Comment:
This is mostly a re-implementation of `_debug_actions`. You can probably just add `session_id` to the logic and reuse it.